pkg/kotsclient/release.go

```diff
@@ -0,0 +1,227 @@
+package kotsclient
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/replicated/pkg/graphql"
+	"github.com/replicatedhq/replicated/pkg/types"
+	"github.com/replicatedhq/replicated/pkg/util"
+)
+
+type GraphQLResponseKotsCreateRelease struct {
+	Data   KotsCreateReleaseData `json:"data,omitempty"`
+	Errors []graphql.GQLError    `json:"errors,omitempty"`
+}
+
+type KotsCreateReleaseData struct {
+	KotsReleaseData KotsReleaseSequence `json:"createKotsRelease"`
+}
+
+type KotsReleaseSequence struct {
+	Sequence int64 `json:"sequence"`
+}
+
+type GraphQLResponseKotsUpdateRelease struct {
+	Data   KotsUpdateReleaseData `json:"data,omitempty"`
+	Errors []graphql.GQLError    `json:"errors,omitempty"`
+}
+
+type KotsUpdateReleaseData struct {
+	KotsReleaseData KotsReleaseSequence `json:"createKotsRelease"`
+}
+
+type GraphQLResponseListReleases struct {
+	Data   *KotsReleasesData  `json:"data,omitempty"`
+	Errors []graphql.GQLError `json:"errors,omitempty"`
+}
+
+type KotsReleasesData struct {
+	KotsReleases []*KotsRelease `json:"allKotsReleases"`
+}
+
+type KotsRelease struct {
+	ID           string         `json:"id"`
+	Sequence     int64          `json:"sequence"`
+	CreatedAt    string         `json:"created"`
+	ReleaseNotes string         `json:"releaseNotes"`
+	Channels     []*KotsChannel `json:"channels"`
+}
+
+type GraphQLResponseUpdateKotsRelease struct {
+	Data *KotsReleaseUpdateData `json:"data,omitempty"`
+}
+
+type KotsReleaseUpdateData struct {
+	UpdateKotsRelease *UpdateKotsRelease `json:"updateKotsRelease"`
+}
+
+type UpdateKotsRelease struct {
+	ID     string `json:"id"`
+	Config string `json:"spec,omitempty"`
+}
+
+func (c *GraphQLClient) CreateRelease(appID string, multiyaml string) (*types.ReleaseInfo, error) {
+	response := GraphQLResponseKotsCreateRelease{}
+
+	request := graphql.Request{
+		Query: `
+		mutation createKotsRelease($appId: ID!, $spec: String!) {
+			createKotsRelease(appId: $appId, spec: $spec) {
+				sequence
+			}
+		}`,
+		Variables: map[string]interface{}{
+			"appId": appID,
+			"spec":  multiyaml,
+		},
+	}
+
+	if err := c.ExecuteRequest(request, &response); err != nil {
+		return nil, err
+	}
+
+	releaseInfo := types.ReleaseInfo{
+		AppID:    appID,
+		Sequence: response.Data.KotsReleaseData.Sequence,
+	}
+
+	return &releaseInfo, nil
+}
+
+var updateKotsRelease = `
```

Comment:
is this used?

Reply:
oh I could, that had been the intention - I set the variable then just pasted the mutation directly into the UpdateRelease function.  <facepalm>.  will fix to as originally intended.